### PR TITLE
test(SignupTest): add test for the signup flow

### PR DESCRIPTION
**Pull request by KurisuLim**
#### Ticket 3. Automate Sign Up scenarios
1. Sign Up Success. (Please make sure you can run your Sign Up test more than 2 times !)
2. Sign Up Wrong credentials (not existing email)
3. Sign Up Empty credentials

### CODE REVIEW

File: src/test/java/pages/BasePage.java [hunk 1]
```diff
@@ -41,4 +42,9 @@ public void assertEquals(By element, String expectText){
         Assert.assertEquals(getText(element), expectText);
     }
 
+    public void assertTrue(By element){
+        waitVisibility(element);
+        Assert.assertTrue(driver.findElement(element).isDisplayed());
```
Comment (mobilman89) [inline src/test/java/pages/BasePage.java:47]: Asserts better use in test cases!

File: src/test/java/pages/SignupPage.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+public class SignupPage extends BasePage {
+
+    public SignupPage(WebDriver driver) {
+        super(driver);
+    }
+
+    //element
+    By firstName = By.cssSelector("[formcontrolname=firstName]");
+    By firstNameError = By.cssSelector("[errormsg=Please enter your First name]");
```
Comment (mobilman89) [inline src/test/java/pages/SignupPage.java]: make all locators as a private

File: src/test/java/pages/SignupPage.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+    By lastName = By.cssSelector("[formcontrolname=lastName]");
+    By lastNameError = By.cssSelector("[errormsg=Please enter your Last name]");
+
+    By businessName = By.cssSelector("[formcontrolname=businessName]");
+    By businessError = By.cssSelector("[errormsg=Please enter your Business name]");
+
+    By phone = By.cssSelector("[formcontrolname=phone]");
+    By phoneError = By.cssSelector("div.card-content div:nth-child(4) div app-field-error-display div div small");
```
Comment (mobilman89) [inline src/test/java/pages/SignupPage.java]: Are you sure that it is the best locator?

File: src/test/java/pages/SignupPage.java [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+    By email = By.cssSelector("[formcontrolname=email]");
+    By emailError = By.cssSelector("div.card-content div:nth-child(5) div app-field-error-display div div small");
+
+    By password = By.cssSelector("[formcontrolname=password]");
+    By passwordError = By.cssSelector("div.card-content div:nth-child(6) div app-field-error-display div div small");
```
Comment (mobilman89) [inline src/test/java/pages/SignupPage.java]: Try to find shorter locator

File: src/test/java/pages/SignupPage.java [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+
+    By privacyCheckbox = By.cssSelector("div.card-content div:nth-child(8) div p-checkbox");
+
+    By createAccBtn = By.cssSelector("button.btn-danger");
+    By alert = By.cssSelector(".alert");
+
+    public SignupPage emptyCredentials(){
```
Comment (mobilman89) [inline src/test/java/pages/SignupPage.java:41]: you can create a User class with fields: firstName, lastName, businessName, phone, email, password like you created a Faker class! and create 3 users with different values(emptyUser, invalidUser, fakerUser) and implement only one method signUp which accept User as a parameter. It allows you avoid a code duplicate

File: src/test/java/tests/SignupTest.java [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+public class SignupTest extends BaseTest{
+    /*
+    * Ticket 3. Automate Sign Up scenarios
+      1. Sign Up Success. (Please make sure you can run your Sign Up test more than 2 times !)
+      2. Sign Up Wrong credentials (not existing email)
+      3. Sign Up Empty credentials
+    *
+    * */
+
+    @Test (priority = 0)
```
Comment (mobilman89) [inline src/test/java/tests/SignupTest.java]: remove priority = 0 please

File: src/test/java/tests/SignupTest.java [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+        .emptyCredentials()
+        .getCurrentUrl("https://testkwidos.tk/auth/register/contractor");
+
+    }
+
+    @Test (priority = 1)
+    public void wrongCredentials(){
+        // The Create Account Button is disable
+        // so basically all field should be correct
+        HomePage homePage = new HomePage(driver);
```
Comment (mobilman89) [inline src/test/java/tests/SignupTest.java]: use one HomePage  instance for all test cases.

File: src/test/java/tests/SignupTest.java [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+                .signupError();
+    }
+
+    @Test (priority = 2)
+    public void successSignup(){
+        HomePage homePage = new HomePage(driver);
+        homePage.testkwidos()
+                .toSignupPage()
+                .fakerSignup()
+                .successAlert();
```
Comment (mobilman89) [inline src/test/java/tests/SignupTest.java]: asserts should placing here

File: src/test/java/tests/SignupTest.java [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+        homePage.testkwidos()
+                .toSignupPage()
+                .wrongSignUp()
+                .signupError();
+    }
+
+    @Test (priority = 2)
+    public void successSignup(){
+        HomePage homePage = new HomePage(driver);
+        homePage.testkwidos()
```
Comment (mobilman89) [inline src/test/java/tests/SignupTest.java]: use @BeforeTest

File: src/test/java/tests/SignupTest.java [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+
+    @Test (priority = 0)
+    public void emptyCredentials(){
+        // The Create Account Button is disable
+        // so there is no way to test an empty state
+        HomePage homePage = new HomePage(driver);
+        homePage.testkwidos()
+        .toSignupPage()
+        .emptyCredentials()
+        .getCurrentUrl("https://testkwidos.tk/auth/register/contractor");
```
Comment (mobilman89) [inline src/test/java/tests/SignupTest.java]: getCurrentUrl method should return a currentUrl and not compare two different urls. Please, move AssertEquals here.

File: src/test/java/pages/SignupPage.java [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+        String currentUrl = driver.getCurrentUrl();
+        Assert.assertEquals(currentUrl, url);
+
+        return this;
+    }
+
+    public SignupPage fakerSignup(){
+        Faker faker = new Faker();
+
+        String fakeFirstName = faker.name().firstName();
```
Comment (mobilman89) [inline src/test/java/pages/SignupPage.java]: these variables not need. You can just pass faker.name().firstName() to setText method

File: src/test/java/pages/SignupPage.java [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+        String fakeLastName = faker.name().lastName();
+        String fakeBusinessName = faker.company().industry();
+        String fakePhoneNumber = faker.phoneNumber().phoneNumber();
+        String fakeEmailAddress = faker.name().firstName();
+        String fakePassword = faker.internet().password();
+
+        setText(firstName, fakeFirstName);
+        setText(lastName, fakeLastName);
```
Comment (mobilman89) [inline src/test/java/pages/SignupPage.java]: setText(lastName, faker.name().lastName());

File: src/test/java/pages/SignupPage.java (hunk 3 above)
Comment (KurisuLim) [reply src/test/java/pages/SignupPage.java]: Yes.

File: src/test/java/pages/SignupPage.java (hunk 4 above)
Comment (KurisuLim) [reply src/test/java/pages/SignupPage.java]: @mobilman89
The html code they used is like string of web.

File: src/test/java/pages/BasePage.java (hunk 1 above)
Comment (KurisuLim) [reply src/test/java/pages/BasePage.java:47]: Its already done.

**Pull request merged**